src/cascadia/TerminalSettingsModel/MatchProfilesEntry.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "MatchProfilesEntry.h"
+#include "JsonUtils.h"
+
+#include "MatchProfilesEntry.g.cpp"
+
+using namespace Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::Settings::Model::implementation;
+
+static constexpr std::string_view NameKey{ "name" };
+static constexpr std::string_view CommandlineKey{ "commandline" };
+static constexpr std::string_view SourceKey{ "source" };
+
+MatchProfilesEntry::MatchProfilesEntry() noexcept :
+    MatchProfilesEntryT<MatchProfilesEntry, ProfileCollectionEntry>(NewTabMenuEntryType::MatchProfiles)
+{
+}
+
+Json::Value MatchProfilesEntry::ToJson() const
+{
+    auto json = NewTabMenuEntry::ToJson();
+
+    JsonUtils::SetValueForKey(json, NameKey, _Name);
+    JsonUtils::SetValueForKey(json, CommandlineKey, _Commandline);
+    JsonUtils::SetValueForKey(json, SourceKey, _Source);
+
+    return json;
+}
+
+winrt::com_ptr<NewTabMenuEntry> MatchProfilesEntry::FromJson(const Json::Value& json)
+{
+    auto entry = winrt::make_self<MatchProfilesEntry>();
+
+    JsonUtils::GetValueForKey(json, NameKey, entry->_Name);
+    JsonUtils::GetValueForKey(json, CommandlineKey, entry->_Commandline);
+    JsonUtils::GetValueForKey(json, SourceKey, entry->_Source);
+
+    return entry;
+}
+
+bool MatchProfilesEntry::MatchesProfile(const Model::Profile& profile)
+{
+    auto isMatching = false;
+
+    if (!_Name.empty())

```

Comment:
So this is now a union of all the properties, not the intersection. I feel like we should probably make it the conjunction. If it's a disjunction, there's no way to filter on multiple properties at once. If it's an intersection instead, then you can still get a union by just adding one entry per property

Reply:
Since it's now a separate function, I'll write a quick fix for it tonight.

Reply:
Luckily for you, European tonight is already now ;) It should now have conjunction semantics.